lib/membrane_rtmp_plugin/rtmp/source/tcp_server.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Membrane.RTMP.Source.TcpServer do
+  @moduledoc """
+  A simple tcp server, which executes given function for each new connection.
+  """
+  use Task
+
+  @enforce_keys [:port, :tcp_options, :serve_fn]
+
+  defstruct @enforce_keys
+
+  @spec start_link(%__MODULE__{}) :: {:ok, pid}
```

Comment:
```suggestion
  @type t  :: %__MODULE__{
    port: :inet.port_number(),
    options: [:inet.inet_backend() | :gen_tcp.listen_option()],
    socker_handler:  (:gen_tcp.socket() -> {:ok, pid} | {:error, reason :: any()})
  }
  
  @spec start_link(t()) :: {:ok, pid}
```

In general, try to put as much specs as you can.

Also the module's doc could be more explanatory, for instance what the `serve_fn` is supposed to be used for (also the name could be changed to e.g. `socker_handler` or something).

Reply:
I renamed the `serve_fn` and added example.